### ISSUE TITLE
[tengine-2.1.2] reqstat Segmentation fault

### ISSUE THREAD

**Issue by annProg** [alibaba/tengine#696]
http段配置
```
req_status_zone reqstat "$upstream_addr,$host,$uri" 10M;
req_status reqstat;
```

server配置
```
server {
	listen 8000;
	server_name localhost;
	location /us {
		req_status_show;
		req_status_show_field req_total http_5xx;
	}
}
```

然后就报 Segmentation fault
```
[root@localhost conf]# ../sbin/nginx -t
Segmentation fault (core dumped)
```

dmesg内容：
```
nginx[679]: segfault at 0 ip 0000000000491490 sp 00007ffe25d62590 error 4 in nginx[400000+b8000]
nginx[4603]: segfault at 0 ip 0000000000491490 sp 00007fff4cca1c30 error 4 in nginx[400000+b8000]
nginx[4958]: segfault at 0 ip 0000000000491490 sp 00007ffe03677f30 error 4 in nginx[400000+b8000]
nginx[12708]: segfault at 0 ip 0000000000491490 sp 00007ffe9db7c0c0 error 4 in nginx[400000+b8000]
nginx[12730]: segfault at 0 ip 0000000000491490 sp 00007ffdcd372a50 error 4 in nginx[400000+b8000]
```

**Comment by cfsego** [comment]
thanks for your report.
this bug was fixed in 2.1.3.
And you may refer to 2.1.3 branch for the latest codes.

**Comment by Colstuwjx** [comment]
+1
